app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/VideoViewModel.java

```diff
@@ -0,0 +1,71 @@
+package com.iitp.njack.iitp_connect.core.youtube;
+
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+import android.content.Context;
+import android.databinding.ObservableInt;
+import android.view.View;
+import com.google.api.client.googleapis.extensions.android.gms.auth.GoogleAccountCredential;
+import com.iitp.njack.iitp_connect.R;
+import java.util.List;
+
+public class VideoViewModel extends ViewModel {
+    public ObservableInt loading;
+    public ObservableInt showEmpty;
+    Context context;
+    private MutableLiveData<YoutubeVideo> selected;
+    private VideoRepository videorepository;
+    private VideoAdapter videoAdapter;
+    private GoogleAccountCredential googleAccountCredential;
+
+    void init(Context context) {
+        this.context = context;
```

Comment:
Use constructor and dependency injection

Reply:
Context is an Android dependent class. It should not be in the ViewModel otherwise it will lead to huge memory leaks.